### PR TITLE
Change of CustomMetric to AutoMetric

### PR DESCRIPTION
**Pull request by chandnisingh**
This needs to go into 3.0

### CODE REVIEW

File: api/src/main/java/com/datatorrent/api/StreamCodec.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
  */
 public interface StreamCodec<T>
 {
-  /**
+  /**                                                 N
```
Comment (243826) [inline api/src/main/java/com/datatorrent/api/StreamCodec.java]: what's this?

File: engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java [hunk 2]
```diff
@@ -429,10 +430,10 @@ public boolean isGatewayConnected()
     }
 
     @Override
-    public Map<String, Object> getCustomMetrics()
+    public Map<String, Object> getMetrics()
     {
       if (StreamingAppMasterService.this.dnmgr != null) {
-        return (Map)StreamingAppMasterService.this.dnmgr.getCustomMetrics();
+        return (Map)StreamingAppMasterService.this.dnmgr.getLatestLogicalMetrics();
```
Comment (243826) [inline engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java]: does this not give a warning about ray types?

File: engine/src/main/java/com/datatorrent/stram/appdata/AppDataPushAgent.java [hunk 3]
```diff
@@ -251,17 +238,17 @@ private JSONObject extractFields(Object o)
...
   }
 
-  private JSONObject getCustomMetricsSchemaData(LogicalPlan.OperatorMeta operatorMeta, Map<String, Object> aggregates)
+  private JSONObject getMetricsSchemaData(LogicalPlan.OperatorMeta operatorMeta, Map<String, Object> aggregates)
   {
     JSONObject result = new JSONObject();
     try {
-      result.put("type", CUSTOM_METRICS_SCHEMA);
-      result.put("version", CUSTOM_METRICS_SCHEMA_VERSION);
+      result.put("type", METRICS_SCHEMA);
```
Comment (243826) [inline engine/src/main/java/com/datatorrent/stram/appdata/AppDataPushAgent.java]: Guys, please clean up these magic values sometime soon.

File: engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java (hunk 2 above)
Comment (chandnisingh) [inline engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java]: I didn't originally write this code. Just changed the name.

**Comment by 243826** [comment]
Also committed to release-3.0 branch.

**Pull request merged**